editor/editor_dock_manager.cpp

```diff
@@ -461,11 +467,11 @@ void EditorDockManager::save_docks_to_config(Ref<ConfigFile> p_layout, const Str
 		window_dump["window_screen"] = wrapper->get_window_screen();
 		window_dump["window_screen_rect"] = DisplayServer::get_singleton()->screen_get_usable_rect(screen);
 
-		String name = dock->get_name();
+		String name = dock->get_display_title();
```

Comment:
We should not use the display title to save/load data. It's now pretty easy to change the title at any time for a dock, so this value cannot be relied on. Saving and loading needs this to be the same or it will not work.

Reply:
The previous code looks like it used the control's name and just ignored the title, since they always effectively had a name before.
I thought it was even encouraged now because the `tab_style_changed` signal emits when the title changes so it updates properly. 
And though it's not a dock, the Debugger bottom panel changes it's display to `Debugger (1)` to show the error count dynamically.
We could just have another property to get a name for saving, and if it is empty fall back to the display title.